server/src/main/java/io/crate/execution/engine/indexing/ShardingUpsertExecutor.java

```diff
@@ -351,7 +358,18 @@ private void maybeSetInterrupt(@Nullable Exception failure) {
 
         @Override
         public final boolean test(ShardedRequests<?, ?> shardedRequests) {
-            return shardedRequests.usedMemoryEstimate() > maxBytesUsableByShardedRequests;
+            long usedMemoryEstimate = shardedRequests.ramBytesUsed();
+            boolean requestsTooBig = usedMemoryEstimate > maxBytesUsableByShardedRequests
+                || usedMemoryEstimate > MAX_MEMORY_PER_REQUEST_IN_BYTES;
```

Comment:
Not sure about this. The additional 20% or at most 100MB is not strictly necessary.

But taking up several GB for bulk requests seems too much.
Or maybe the 20% is still too aggressive and we should go with something like 5% or 10%:

With 10%

```
128.0 GB   → 12800 MB
16.0 GB    →  1600 MB
 8.0 GB    →   800 MB
 2.0 GB    →   200 MB
 0.5 GB    →    50 MB
```


With 5%

```
128.0 GB    →  6400 MB
 16.0 GB    →   800 MB
  8.0 GB    →   400 MB
  2.0 GB    →   100 MB
  0.5 GB    →    25 MB
```

Reply:
As the 100MB can be reached quite fast, I'd tend more to lower the percentage to `10%` to allow more on higher HEAP configurations. Or would you still keep the `100MB` hard limit even with a lower percentage based threshold?